oqsprov/oqs_encode_key2any.c

```diff
@@ -800,30 +802,87 @@ static int oqsx_pki_priv_to_der(const void *vxkey, unsigned char **pder) {
                 version = V_ASN1_UNDEF;
                 pval = NULL;
             }
-            if (!PKCS8_pkey_set0(p8inf_internal, OBJ_nid2obj(nid), 0, version,
-                                 pval, buf, buflen)) {
-                for (int j = 0; j <= i; j++) {
-                    OPENSSL_cleanse(aString[j]->data, aString[j]->length);
-                    ASN1_OCTET_STRING_free(aString[j]);
-                    OPENSSL_cleanse(aType[j]->value.sequence->data,
-                                    aType[j]->value.sequence->length);
-                    OPENSSL_clear_free(temp[j], templen[j]);
+            if (nid == EVP_PKEY_ED25519 || nid == EVP_PKEY_ED448) {
```

Comment:
I can't claim to understand why special ed25519 and ed448 handling is required now (?), nor how this code achieves this but in order to not stand in the way of progress (and to keep supporting #317), I'm fine with this for now -- but added a line item to review this in #375.

Reply:
The special cases for ed relate to the double octet encoding I need to do for the private key, according with RFC 8410.
This relates to #466.